projects/cm_mcu/LocalTasks.c

```diff
@@ -961,6 +1020,22 @@ void init_registers_ff()
   apollo_i2c_ctl_reg_w(4, 0x21, 1, 0x02, 1, 0x00); // 00000000 [P07..P00]
   apollo_i2c_ctl_reg_w(4, 0x21, 1, 0x03, 1, 0x01); // 00000011 [P17..P10]
 
+  // 7b) U6 default output values (I2C address 0x22 on I2C channel #3)
+  // The outputs on P10 and P11 should default to "1".
+  // This negates the active-lo "RESET" inputs on the VU7P FireFlys.
+  // The outputs on P12 should default to "0" to enable the optical output.
+  // The output on P13 should default to "0" until determined otherwise.
+  // # set third I2C switch on channel 3 (U4, address 0x72) to port 2
+  apollo_i2c_ctl_w(4, 0x72, 1, 0x04);
+  apollo_i2c_ctl_reg_w(4, 0x21, 1, 0x02, 1, 0x00); // 00000000 [P07..P00]
+  apollo_i2c_ctl_reg_w(4, 0x21, 1, 0x03, 1, 0x03); // 00000011 [P17..P10]
+
+  xSemaphoreGive(ffldaq_f1_args.xSem); // if we have a semaphore, give it
```

Comment:
please use the i2c4_sem variable directly here (it's clearer). Ditto in all the other cases too.

Reply:
Done (as well as other cases)